wandb/sdk/launch/runner/gcp_vertex.py

```diff
@@ -0,0 +1,189 @@
+import os
+import json
+import subprocess
+from typing import Any, Dict, List, Optional
+import datetime
+import yaml
+
+from six.moves import shlex_quote
+
+import wandb
+from wandb.sdk.launch.docker import validate_docker_installation
+from wandb.errors import CommError, LaunchError
+from wandb.apis.internal import Api
+
+from google.cloud import aiplatform
+
+from .._project_spec import (
+    DEFAULT_LAUNCH_METADATA_PATH,
+    get_entry_point_command,
+    LaunchProject,
+)
+from ..docker import (
+    build_docker_image_if_needed,
+    construct_gcp_image_uri,
+    docker_image_exists,
+    docker_image_inspect,
+    generate_docker_base_image,
+    pull_docker_image,
+    validate_docker_installation,
+)
+from .abstract import AbstractRun, AbstractRunner, Status
+from ..utils import (
+    PROJECT_DOCKER_ARGS,
+    PROJECT_SYNCHRONOUS,
+)
+
+
+class VertexSubmittedRun(AbstractRun):
+    def __init__(self, model) -> None:
+        self._model = model
+
+    @property
+    def id(self) -> int:
+        pass
+
+    def wait(self) -> bool:
+        pass
+
+    def get_status(self) -> Status:
+        # todo: only sync for now so always finished
+        return Status("finished")
+
+    def cancel(self) -> None:
+        pass
+
+
+class VertexRunner(AbstractRunner):
+    def __init__(self, *args, **kwargs) -> None:
+        super().__init__(*args, **kwargs)
+
+    def run(self, launch_project: LaunchProject) -> Optional[AbstractRun]:
+        resource_args = launch_project.resource_args
+        gcp_config = get_gcp_config(resource_args.get("gcp_config") or "default")
```

Comment:
What if the user passes in the account info via resource args?

Reply:
this is used to fetch defaults, eg the lines after this allow the user to pass in project/zone etc via resource args and it defaults to whatever's in the gcloud config otherwise